### ISSUE TITLE
loss:nan

### ISSUE THREAD

**Issue by sudacn** [zingp/pointer-generator-pytorch#2]
你好，我尝试跑你的代码，刚开始loss在6-2徘徊 60K step后出现loss:nan，尝试把lr改了0.1 还是没有解决  请问作者遇到了？后期怎么解决。感谢

**Comment by yamonc** [comment]
你好，请问你的问题解决了吗？我的是刚开始训练就出现了nan。

**Comment by yamonc** [comment]
> 你好，我尝试跑你的代码，刚开始loss在6-2徘徊 60K step后出现loss:nan，尝试把lr改了0.1 还是没有解决 请问作者遇到了？后期怎么解决。感谢
你好，请问你的问题解决了吗？我的是刚开始训练就出现了nan。

**Comment by zingp** [comment]
你好，可以打印梯度和损失看下，我第一感觉是学习率太大了，可以把Adam的学习率调整至0.001，试试




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zingp/pointer-generator-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月6日(星期二) 上午10:40
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [zingp/pointer-generator-pytorch] loss:nan (#2)





  
你好，我尝试跑你的代码，刚开始loss在6-2徘徊 60K step后出现loss:nan，尝试把lr改了0.1 还是没有解决 请问作者遇到了？后期怎么解决。感谢
 你好，请问你的问题解决了吗？我的是刚开始训练就出现了nan。

**Comment by yamonc** [comment]
> 你好，可以打印梯度和损失看下，我第一感觉是学习率太大了，可以把Adam的学习率调整至0.001，试试
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "zingp/pointer-generator-pytorch" ***@***.***&gt;; 发送时间:&nbsp;2021年4月6日(星期二) 上午10:40 ***@***.***&gt;; ***@***.***&gt;; 主题:&nbsp;Re: [zingp/pointer-generator-pytorch] loss:nan (#2) 你好，我尝试跑你的代码，刚开始loss在6-2徘徊 60K step后出现loss:nan，尝试把lr改了0.1 还是没有解决 请问作者遇到了？后期怎么解决。感谢 你好，请问你的问题解决了吗？我的是刚开始训练就出现了nan。 — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.
---------------------------------------------
您好，还是出现了：
steps 29000, seconds for 1000 steps: 88.34, loss: nan
--------------
我的超参数：
import os
import torch
from numpy import random

root_dir = "data_demo/finished_files"
train_data_path = os.path.join(root_dir, "chunked/train_*")
eval_data_path = os.path.join(root_dir, "val.bin")
decode_data_path = os.path.join(root_dir, "val.bin")
vocab_path = os.path.join(root_dir, "vocab")
log_root = "logs/weibo_adagrad"
if not os.path.isdir(log_root):
    os.makedirs(log_root)


hidden_dim = 256
emb_dim = 128
batch_size = 16

max_enc_steps = 200

max_dec_steps = 40
beam_size = 4

min_dec_steps = 20
vocab_size = 50_000

lr = 0.15
adagrad_init_acc = 0.1
rand_unif_init_mag = 0.02
trunc_norm_init_std = 1e-4
max_grad_norm = 2.0

pointer_gen = True

is_coverage = True

cov_loss_wt = 1.0

eps = 1e-12
max_iterations = 750_000

lr_coverage=0.15


use_gpu = True
GPU = "cuda:0"
USE_CUDA = use_gpu and torch.cuda.is_available()     # 是否使用GPU
NUM_CUDA = torch.cuda.device_count()
DEVICE = torch.device(GPU if USE_CUDA else 'cpu')

SEED = 1234
random.seed(SEED)
torch.manual_seed(SEED)
if USE_CUDA:

    torch.manual_seed(SEED)
环境用的是pytorch1.5,tensorflow2.2.0的，ubuntu系统跑的。
请问怎么解决。

**Comment by yamonc** [comment]
模型中的decode中有一个attention_network,是不是也得需要梯度裁剪？

**Comment by zingp** [comment]
我太久没有看这个网络的代码了，通常每个部分都需要梯度裁剪，防止梯度过大。




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "zingp/pointer-generator-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年4月6日(星期二) 下午3:48
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [zingp/pointer-generator-pytorch] loss:nan (#2)





 
模型中的decode中有一个attention_network,是不是也得需要梯度裁剪？

**Comment by yamonc** [comment]
> 我太久没有看这个网络的代码了，通常每个部分都需要梯度裁剪，防止梯度过大。
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "zingp/pointer-generator-pytorch" ***@***.***&gt;; 发送时间:&nbsp;2021年4月6日(星期二) 下午3:48 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [zingp/pointer-generator-pytorch] loss:nan (#2) 模型中的decode中有一个attention_network,是不是也得需要梯度裁剪？ — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

好的。谢谢您的回复。

**Comment by yamonc** [comment]
> 我太久没有看这个网络的代码了，通常每个部分都需要梯度裁剪，防止梯度过大。
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "zingp/pointer-generator-pytorch" ***@***.***&gt;; 发送时间:&nbsp;2021年4月6日(星期二) 下午3:48 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [zingp/pointer-generator-pytorch] loss:nan (#2) 模型中的decode中有一个attention_network,是不是也得需要梯度裁剪？ — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

大佬，您好，我把您的算法模型那块换了，换成transformer和指针结合的了，然后训练了50w次，现在修改了config中的参数（is_coverage=True）,正在接着训练好的模型训练。我想问一下，我跑完了之后，我看见run.sh里面有eval和decode的操作，所以想问一下，这两个有什么区别吗？目前知道的好像decode用到了beam算法，这两个文件是干什么用的，执行顺序又是怎样的？望回复，谢谢大佬。（ps，说的有点乱，总的来说就是eval.py和decode.py文件的用处是什么，两者执行循序，还有就是哪个文件可以测出rouge分数。）
